TarSCM/scm/base.py

```diff
@@ -249,4 +276,5 @@ def unlock_cache(self):
             self.lock_file = None
 
     def finalize(self):
+        self.cleanup()
```

Comment:
This also looks like a bugfix which is not related to proxy, correct?

Reply:
No. This will call the corresponding cleanup() method for each SCM. Some of them require creating temporary directories/config files which must be removed after it's done.